### PR TITLE
refactor: Migrate Note Fragment to compose

### PR DESCRIPTION
**Pull request by itsPronay**
jira: [MIFOSAC-187 ](https://mifosforge.jira.com/browse/MIFOSAC-187)

xml | compose|
-----| -----
image | image

- [ ] Apply the `MifosStyle.xml` style template to your code in Android Studio.

- [ ] Run the unit tests with `./gradlew check` to make sure you didn't break anything

- [ ] If you have multiple commits please combine them into one commit by squashing them.

### CODE REVIEW

File: feature/note/src/main/java/com/mifos/feature/note/NoteScreen.kt [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+        snackbarHostState = snackBarHostState
+    ) { paddingValues ->
+        Box(
+            modifier = Modifier
+                .fillMaxSize()
+                .padding(paddingValues)
+                .nestedScroll(pullRefreshState.nestedScrollConnection)
+        ) {
+            when (uiState) {
+                NoteUiState.Initial -> Unit
```
Comment (Aditya-gupta99) [inline feature/note/src/main/java/com/mifos/feature/note/NoteScreen.kt:97]: Remove this State

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/online/note/NoteFragment.kt [hunk 2]
```diff
@@ -21,115 +24,33 @@ import javax.inject.Inject
...
-                }
-
-                is NoteUiState.ShowProgressbar -> showProgressbar(true)
-                is NoteUiState.ShowResetVisibility -> {
-                    showProgressbar(false)
-                    showResetVisibility()
-                }
+        toolbar?.visibility = View.GONE
+        return ComposeView(requireContext()).apply {
+            setContent {
```
Comment (Aditya-gupta99) [inline mifosng-android/src/main/java/com/mifos/mifosxdroid/online/note/NoteFragment.kt:47]: setViewCompositionStrategy(ViewCompositionStrategy.DisposeOnViewTreeLifecycleDestroyed)
add this line above setContent

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/online/note/NoteRepositoryImp.kt [hunk 3]
```diff
@@ -8,10 +9,10 @@ import javax.inject.Inject
...
  */
-class NoteRepositoryImp @Inject constructor(private val dataManagerNote: DataManagerNote) :
-    NoteRepository {
-
-    override fun getNotes(entityType: String?, entityId: Int): Observable<List<Note>> {
-        return dataManagerNote.getNotes(entityType, entityId)
-    }
-}
\ No newline at end of file
+//class NoteRepositoryImp @Inject constructor(private val dataManagerNote: DataManagerNote) :
```
Comment (Aditya-gupta99) [inline mifosng-android/src/main/java/com/mifos/mifosxdroid/online/note/NoteRepositoryImp.kt:12]: Delete this repo

**Review by Aditya-gupta99 — CHANGES_REQUESTED** [review]

**Review by Aditya-gupta99 — APPROVED** [review]
LGTM

**Review by therajanmaurya — APPROVED** [review]

**Pull request merged**